plugins/destination/plugin.go

```diff
@@ -12,50 +11,35 @@ import (
 	"github.com/rs/zerolog"
 )
 
-type writerType int
-
-const (
-	unmanaged writerType = iota
-	managed
-)
-
-const (
-	defaultBatchTimeoutSeconds = 20
-	defaultBatchSize           = 10000
-	defaultBatchSizeBytes      = 5 * 1024 * 1024 // 5 MiB
-)
-
 type NewClientFunc func(context.Context, zerolog.Logger, specs.Destination) (Client, error)
 
-type ManagedWriter interface {
-	WriteTableBatch(ctx context.Context, table *schema.Table, data []arrow.Record) error
-}
-
-type UnimplementedManagedWriter struct{}
-
-var _ ManagedWriter = UnimplementedManagedWriter{}
-
-func (UnimplementedManagedWriter) WriteTableBatch(context.Context, *schema.Table, []arrow.Record) error {
-	panic("WriteTableBatch not implemented")
-}
-
 type UnmanagedWriter interface {
-	Write(ctx context.Context, tables schema.Tables, res <-chan arrow.Record) error
+	Write(context.Context, specs.Source, schema.Tables, time.Time, <-chan arrow.Record) error
```

Comment:
These (`Source` and `Time`) were added to both writer signatures to somewhat unify how we push the data (only difference being a channel vs. a slice) and could be useful eventually. Also helps us to [remove](https://github.com/cloudquery/plugin-sdk/pull/870/files#diff-e8ee472a920c34b2e1ac3e57c75020a57a5ec9691465ebee51927e3496bfa9a6) `unmanaged_writer.go`

Reply:
I don't think passing source spec to the destination os a good idea.
IMO, it's the source job to provide the data and the destination has to store it without looking at what the source spec was.